packages/terra-abstract-modal/src/AbstractModal.jsx

```diff
@@ -31,6 +31,10 @@ const propTypes = {
    * If set to true, the modal will close when a mouseclick is triggered outside the modal.
    */
   closeOnOutsideClick: PropTypes.bool,
+  /**
+   * If set to true, the Escape key will trigger deactivation of the focus trap.
+   */
+  escapeDeactivates: PropTypes.bool,
```

Comment:
I don't think we need a prop to enable this. There isn't a scenario where we'd want the modal to be open and the focus trap to be disabled, right?

Reply:
Here where a few scenarios I thought where we need a modal open but the focus to not be within the modal focus trap.

* Notification Dialog open above a modal
* Popup / Select / Datepicker open inside a modal (shift focus to these which are portalled outside of the modal DOM/focus trap)

I haven't thought through what impact the ecapseDeactive functionality would have on those scenarios much. I think we have test cases for these we could test and see if having it always be default works. If so, I'm fine with removing the prop.

Reply:
It's mostly a personal opinion, but I really like to use the is syntax when dealing with boolean variables. For this instance, I think isDeactivatedByEscape or isDeactivatedByEsc would be great candidate names for that variable.

Reply:
@JakeLaCombe  I actually think we defined the `is` / `has` boolean prop naming as a terra-convention for boolean props. We have it documented here: https://engineering.cerner.com/terra-ui/#/contributing/terra-ui/conventions

Reply:
@tbiethman @bjankord Where did we land on having the prop at all? Yay/nay?

Reply:
I see two solutions:
1. Don't expose a prop and never allow the FocusTrap to deactivate itself.
2. Set the escapeDeactivates option based on our existing closeOnEsc prop, set the onDeactivate option with a function that will call the onRequestClose prop, and remove our own keydown handlers for escape handling. 

I think 2 would allow us to more easily transition to newer versions of the focus-trap-react that support auto-pausing nested focus traps: https://github.com/davidtheclark/focus-trap#one-at-a-time. However, we'd definitely have to investigate the cascading effects of such a change.

Reply:
I talked with @bjankord, and opted for option 1 for the time being. He suggested coming back around to option 2 after investigating the effects of that change.